### PR TITLE
[BugFix] Fix the exception while reading date and datetime type of paimon table

### PR DESCRIPTION
**Pull request by miomiocat**
Fixes #issue

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiColumnValue.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public double getDouble() {
     }
 
     @Override
-    public String getString() {
+    public String getString(ColumnType.TypeValue type) {
```
Comment (dirtysalt) [inline java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiColumnValue.java:78]: how this thing pass compiler check?

Or do we have another method `getString()` ?

File: fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonPredicateConverter.java [hunk 2]
```diff
@@ -171,9 +171,7 @@ public Predicate visitLikePredicateOperator(LikePredicateOperator operator, Void
         if (operator.getLikeType() == LikePredicateOperator.LikeType.LIKE) {
             if (operator.getChild(1).getType().isStringType()) {
                 String literal = ((BinaryString) getLiteral(operator.getChild(1))).toString();
-                if (literal != null && literal.length() > 1 &&
-                        literal.indexOf("%") == literal.length() - 1 &&
-                        literal.charAt(0) != '%') {
+                if (literal.length() > 1 && literal.indexOf("%") == literal.length() - 1 && literal.charAt(0) != '%') {
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonPredicateConverter.java:174]: Do we have any UT to cover this case?

Comment (miomiocat) [reply fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonPredicateConverter.java:174]: PaimonPredicateConverterTest.testBinaryString() covered.

File: java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiColumnValue.java (hunk 1 above)
Comment (miomiocat) [reply java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiColumnValue.java:78]: The signature of `getString()` has changed in this PR.

**Review by dirtysalt — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**